src/test/fuzz/util.cpp

```diff
@@ -7,15 +7,196 @@
 #include <util/rbf.h>
 #include <version.h>
 
-bool FuzzedSock::Wait(std::chrono::milliseconds timeout, Event requested, Event* occurred ) const
+FuzzedSock::FuzzedSock(FuzzedDataProvider& fuzzed_data_provider)
+    : m_fuzzed_data_provider{fuzzed_data_provider}
 {
-    if (!m_fuzzed_data_provider.ConsumeBool()) {
+    m_socket = fuzzed_data_provider.ConsumeIntegral<SOCKET>();
```

Comment:
Not changed in this PR, but is there any scenario where we want `m_socket !=  INVALID_SOCKET` here?

Otherwise I'd prefer going with `INVALID_SOCKET`. Perhaps `Sock::m_socket` could be default initialized to `INVALID_SOCKET`. That should be a safe default.

Reply:
Normally we want the fuzzed socket's `m_socket` to be `!= INVALID_SOCKET`, otherwise a code that does `if (sock.Get() == INVALID_SOCKET)` will "misbehave" when mocked.

Reply:
@vasild Oh, I didn't think about the possibility to peek at `m_socket` via `Get()`. But then I guess we want two cases: 1.)  `m_socket == INVALID_SOCKET`, and 2.) `m_socket != INVALID_SOCKET` where `m_socket` is a very number which is unlikely to coincide with a real opened file descriptor?

The current code allows `m_socket ` to be set to low numbers such as `0`, `1`, `2`, `3`, etc which may be problematic :)

Reply:
Perhaps `m_socket = fuzzed_data_provider.ConsumeIntegralInRange<SOCKET>(INVALID_SOCKET - 1, INVALID_SOCKET)`?

Reply:
Right! And we can do even better:

* add a method to check if the `Sock` object "owns" a socket, similar to [`std::unique_ptr::operator bool()`](https://en.cppreference.com/w/cpp/memory/unique_ptr/operator_bool) so that callers don't do `Get() == INVALID_SOCKET`.
* Add wrapper methods for `getsockname()`, `setsockopt()`, `bind()` and `listen()`. That way more code will be mockable, but more importantly - then nobody would need to call `Sock::Get()` anymore so the value of `m_socket` will remain "hidden" within the `Sock` class.

Reply:
@vasild Sounds good! Until we have that in place I think it makes sense to setting `m_socket` to low numbers here to avoid nasty surprises :)  The thought of random reads/writes to existing open file descriptors is a bit scary even if only from fuzzing code.

Reply:
Low fd values avoided in https://github.com/bitcoin/bitcoin/pull/21677

Reply:
> Add wrapper methods for ...

Done in https://github.com/bitcoin/bitcoin/pull/21700